pallets/estate/src/lib.rs

```diff
@@ -668,6 +674,106 @@ pub mod pallet {
 				},
 			)
 		}
+
+		#[pallet::weight(T::WeightInfo::dissolve_estate())]
+		pub fn dissolve_estate(
+			origin: OriginFor<T>,
+			estate_id: EstateId,
+			metaverse_id: MetaverseId,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				!T::AuctionHandler::check_item_in_auction(ItemId::Estate(estate_id)),
+				Error::<T>::EstateAlreadyInAuction
+			);
+
+			let landUnits = Estates::<T>::get(estate_id).ok_or(Error::<T>::EstateDoesNotExist)?;
+
+			EstateOwner::<T>::try_mutate_exists(&who, &estate_id, |estate_by_owner| {
+				//ensure there is record of the estate owner with estate id and account id
+				ensure!(estate_by_owner.is_some(), Error::<T>::NoPermission);
+
+				// Reset estate ownership
+				*estate_by_owner = None;
+
+				// Remove estate
+				Estates::<T>::remove(&estate_id);
+
+				// Update total estates
+				let total_estates_count = Self::all_estates_count();
+				let new_total_estates_count = total_estates_count
+					.checked_sub(One::one())
+					.ok_or("Overflow adding new count to total estates")?;
+				AllEstatesCount::<T>::put(new_total_estates_count);
+
+				// Remove land units - remove relationship between landunit and estate
+				for landUnit in landUnits.clone() {
+					LandUnits::<T>::remove(metaverse_id, landUnit);
```

Comment:
yeah, not sure a single mutate would work. Initially we were expecting the land units to have the full list of land units. i.e. contain the new one being added, and won't contain the ones being removed. Because we are going to separate this one call into two extrincs, I think problem will be solved.

Reply:
I believe when you dissolve the Estate, you just mutate a list of land units belong to that Estate to the owner.